### PR TITLE
fix(EMS-3524): no pdf - data migration - account, application relationships

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:

This PR fixes a variety of issues with data migration where:

- The `Account` table was missing an `accountStatus` relationship ID.
- The `Application` table was missing an `nominatedLossPayee` relationship ID.
- Various relationship creations would fail due to KeystoneJS `.create()` issues when running the migration script.
- The UUID's created during migration were not aligned with the CUID format used for all other KeystoneJS generated data.
- The `creditPeriodWithBuyer` was missing a VarChar in the KeystoneJS schema.
- The `BuyerTradingHistory` and `Business` tables had missing default currency codes (GBP).

The migration script now executes pure, raw SQL queries, instead of using the KeystoneJS context API.

## TODO
- Double check the database with fresh eyes.

## Resolution :heavy_check_mark:

- Create a new `createCuid` function.
- Add additional migration scripts, move all KeystoneJS related migration functions to use raw SQL queries:
  - `updateAccountStatusColumns`
  - `createAccountStatusRows`
  - `createCompanyDifferentTradingAddress`
  - `updateCompanyDifferentTradingAddress`
  - `updateExportContractAgentServiceCharges`
  - `updateExportContractAgentServices`
  - `updateExportContractAgents`
  - `updateExportContractPrivateMarket`
  - `createInitialLossPayees`
  - `updateLossPayeeFinancialInternationalVector`
  - `updateLossPayeeFinancialUkVector`
- Create new migration "get all" functions:
  - `getAllAccountStatuses`
  - `getAllCompanies`
  - `getAllCompanyDifferentTradingAddresses`
  - `getAllExportContractAgentServiceCharges`
  - `getAllExportContractAgentServices`
  - `getAllExportContractAgents`
  - `getAllExportContracts`
  - `getAllLossPayeeFinancialInternationalVectors`
  - `getAllLossPayeeFinancialInternational`
  - `getAllLossPayeeFinancialUkVectors`
  - `getAllLossPayeeFinancialUk`
  - `getAllLossPayees`
  - `getAllPrivateMarkets`

## Miscellaneous :heavy_plus_sign:

- Fix some typos.
- Update `executeSqlQuery` to reject a call if `connection.query` is not a function.

### CODE REVIEW

File: src/api/data-migration/version-1-to-version-2/create-new-account-status-relationships/account-status-columns.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { AccountStatus } from '../../../types';
+
+/**
+ * updateAccountStatusColumns
+ * Update "account status" columns for all existing accounts.
+ * 1) Map over each account status entry.
+ * 2) Update the Account table to have a status ID/relationship.
+ * @param {Connection} connection: SQL database connection
+ * @param {Array<AccountStatus>} statusRows: Account status entries
+ * @param {Array<AccountMvp>} accounts: MVP accounts
```
Comment (Zainzzkk) [inline src/api/data-migration/version-1-to-version-2/create-new-account-status-relationships/account-status-columns.ts:12]: wrong param type?

File: src/api/data-migration/version-1-to-version-2/create-new-application-relationships/create-company-different-trading-address.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * @param {Array<Application>} applications: Applications
+ * @returns {Promise<Array<object>>} executeSqlQuery responses
+ */
+const createCompanyDifferentTradingAddress = async (connection: Connection, applications: Array<Application>) => {
+  const loggingMessage = 'Creating companyDifferentTradingAddresses entries with company relationships';
+
+  console.info(`✅ ${loggingMessage}`);
+
+  try {
+    const jointlyInsuredPartyPromises = applications.map(async (application: Application) => {
```
Comment (Zainzzkk) [inline src/api/data-migration/version-1-to-version-2/create-new-application-relationships/create-company-different-trading-address.ts:22]: is  this the right const name?

File: src/api/data-migration/version-1-to-version-2/get-all-companies.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ */
+const getAllCompanies = async (connection: Connection) => {
+  const loggingMessage = 'Getting all companies';
+
+  console.info(`✅ ${loggingMessage}`);
+
+  try {
+    const query = 'SELECT * FROM Company';
+
+    const [exportContracts] = await executeSqlQuery({ connection, query, loggingMessage });
```
Comment (Zainzzkk) [inline src/api/data-migration/version-1-to-version-2/get-all-companies.ts:18]: wrong const

File: src/api/data-migration/version-1-to-version-2/get-all-company-different-trading-addresses.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ */
+const getAllCompanyDifferentTradingAddresses = async (connection: Connection) => {
+  const loggingMessage = 'Getting all company different trading addresses';
+
+  console.info(`✅ ${loggingMessage}`);
+
+  try {
+    const query = 'SELECT * FROM CompanyDifferentTradingAddress';
+
+    const [exportContracts] = await executeSqlQuery({ connection, query, loggingMessage });
```
Comment (Zainzzkk) [inline src/api/data-migration/version-1-to-version-2/get-all-company-different-trading-addresses.ts:18]: same as above

File: src/api/data-migration/version-1-to-version-2/get-all-export-contract-agent-service-charges.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ */
+const getAllExportContractAgentServiceCharges = async (connection: Connection) => {
+  const loggingMessage = 'Getting all export contract agent service charges';
+
+  console.info(`✅ ${loggingMessage}`);
+
+  try {
+    const query = 'SELECT * FROM ExportContractAgentServiceCharge';
+
+    const [exportContracts] = await executeSqlQuery({ connection, query, loggingMessage });
```
Comment (Zainzzkk) [inline src/api/data-migration/version-1-to-version-2/get-all-export-contract-agent-service-charges.ts:18]: same as above

File: src/api/data-migration/version-1-to-version-2/get-all-export-contract-agent-services.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+ */
+const getAllExportContractAgentServices = async (connection: Connection) => {
+  const loggingMessage = 'Getting all export contract agent services';
+
+  console.info(`✅ ${loggingMessage}`);
+
+  try {
+    const query = 'SELECT * FROM ExportContractAgentService';
+
+    const [exportContracts] = await executeSqlQuery({ connection, query, loggingMessage });
```
Comment (Zainzzkk) [inline src/api/data-migration/version-1-to-version-2/get-all-export-contract-agent-services.ts:18]: same as above?

File: src/api/data-migration/version-1-to-version-2/get-all-export-contract-agents.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+ */
+const getAllExportContractAgents = async (connection: Connection) => {
+  const loggingMessage = 'Getting all export contract agents';
+
+  console.info(`✅ ${loggingMessage}`);
+
+  try {
+    const query = 'SELECT * FROM ExportContractAgent';
+
+    const [exportContracts] = await executeSqlQuery({ connection, query, loggingMessage });
```
Comment (Zainzzkk) [inline src/api/data-migration/version-1-to-version-2/get-all-export-contract-agents.ts:18]: same as above

File: src/api/data-migration/version-1-to-version-2/get-all-loss-payee-financial-international-vectors.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+ */
+const getAllLossPayeeFinancialUk = async (connection: Connection) => {
+  const loggingMessage = 'Getting all loss payee - financial international vector entries';
+
+  console.info(`✅ ${loggingMessage}`);
+
+  try {
+    const query = 'SELECT * FROM LossPayeeFinancialInternationalVector';
+
+    const [exportContracts] = await executeSqlQuery({ connection, query, loggingMessage });
```
Comment (Zainzzkk) [inline src/api/data-migration/version-1-to-version-2/get-all-loss-payee-financial-international-vectors.ts:18]: same as above

File: src/api/data-migration/version-1-to-version-2/get-all-loss-payee-financial-internationals.ts [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+ */
+const getAllLossPayeeFinancialInternational = async (connection: Connection) => {
+  const loggingMessage = 'Getting all loss payee - financial international entries';
+
+  console.info(`✅ ${loggingMessage}`);
+
+  try {
+    const query = 'SELECT * FROM LossPayeeFinancialInternational';
+
+    const [exportContracts] = await executeSqlQuery({ connection, query, loggingMessage });
```
Comment (Zainzzkk) [inline src/api/data-migration/version-1-to-version-2/get-all-loss-payee-financial-internationals.ts:18]: same as above

File: src/api/data-migration/version-1-to-version-2/get-all-loss-payee-financial-uk-vectors.ts [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+ */
+const getAllLossPayeeFinancialUk = async (connection: Connection) => {
+  const loggingMessage = 'Getting all loss payee - financial UK vector entries';
+
+  console.info(`✅ ${loggingMessage}`);
+
+  try {
+    const query = 'SELECT * FROM LossPayeeFinancialUkVector';
+
+    const [exportContracts] = await executeSqlQuery({ connection, query, loggingMessage });
```
Comment (Zainzzkk) [inline src/api/data-migration/version-1-to-version-2/get-all-loss-payee-financial-uk-vectors.ts:18]: same as above

File: src/api/data-migration/version-1-to-version-2/get-all-loss-payee-financial-uks.ts [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+ */
+const getAllLossPayeeFinancialUk = async (connection: Connection) => {
+  const loggingMessage = 'Getting all loss payee - financial UK entries';
+
+  console.info(`✅ ${loggingMessage}`);
+
+  try {
+    const query = 'SELECT * FROM LossPayeeFinancialUk';
+
+    const [exportContracts] = await executeSqlQuery({ connection, query, loggingMessage });
```
Comment (Zainzzkk) [inline src/api/data-migration/version-1-to-version-2/get-all-loss-payee-financial-uks.ts:18]: same a above

File: src/api/package.json [hunk 12]
```diff
@@ -31,6 +31,7 @@
     "@keystone-6/auth": "^7.0.3",
     "@keystone-6/core": "^5.8.0",
     "@keystone-6/fields-document": "^8.0.2",
+    "@paralleldrive/cuid2": "^2.2.2",
```
Comment (Zainzzkk) [inline src/api/package.json:34]: is this package added to the ticket to remove after migration?

Comment (ttbarnes) [reply src/api/package.json:34]: Yes, EMS-3521

**Pull request merged**